### PR TITLE
Gh-3108: Improve testing for GafferPopVertexProperty

### LINKED ISSUE #3108: Improve testing for GafferPopVertexProperty
Improve the testing for the GafferPopVertexProperty class to at least >80%

### PR DESCRIPTION
**Pull request by tb06904**
Improves the testing on `GafferPopVertexProperty` class to above > 80% coverage.

Slight tweak to reduce mock duplication in the existing test class too.

# Related issue

- Resolve #3108

### CODE REVIEW

File: library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java [hunk 1]
```diff
@@ -150,25 +151,73 @@ void shouldNotAddOrGetIllegalProperties() {
...
+    void shouldOnlyCreateValidGafferPopVertexPropertyObjects() {
+        // Given
+        final GafferPopGraph graph = mock(GafferPopGraph.class);
+        given(graph.features()).willReturn(features);
+        given(features.vertex()).willReturn(vertexFeatures);
+        given(vertexFeatures.properties()).willReturn(vertexPropertyFeatures);
+        final String propValue = "propValue";
+        final String nestedKey = "nestedKey";
+        final String nestedValue = "nestedValue";
+        // Make new vertex with the mocked bits
```
Comment (GCHQDeveloper314) [inline library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java:164]: Not sure these comments are required. Seems self explanatory to me.

File: library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java [hunk 2]
```diff
@@ -150,25 +151,73 @@ void shouldNotAddOrGetIllegalProperties() {
...
+        assertThat(prop.element()).isEqualTo(vertex);
+        assertThat(prop.isPresent()).isTrue();
+        assertThatExceptionOfType(UnsupportedOperationException.class).isThrownBy(() -> prop.remove());
+
+        // Check nested properties work
+        assertThat(prop.property(nestedKey, nestedValue)).isEqualTo(new GafferPopProperty<Object>(vertex, nestedKey, nestedValue));
+        assertThat(prop.keys()).containsExactlyInAnyOrder(nestedKey);
+        assertThat(prop.property(nestedKey)).isEqualTo(new GafferPopProperty<Object>(prop, nestedKey, nestedValue));
+
+        // Check can't make a bad property
```
Comment (GCHQDeveloper314) [inline library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java:192]:
```suggestion
        // Check can't create an invalid property
```

File: library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java [hunk 3]
```diff
@@ -150,25 +151,73 @@ void shouldNotAddOrGetIllegalProperties() {
...
 
+        // Set and get a property
+        vertex.property(Cardinality.list, TestPropertyNames.STRING, "propValue");
+        GafferPopVertexProperty<Object> prop = (GafferPopVertexProperty<Object>) vertex.property(TestPropertyNames.STRING);
+
         // Set the vertex to read only
         vertex.setReadOnly();
 
-        // Attempt to add a property
+        // Attempt to add a properties
```
Comment (GCHQDeveloper314) [inline library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java:216]: Should be this?
```suggestion
        // Attempt to add some properties
```

File: library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java (hunk 1 above)
Comment (tb06904) [reply library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexTest.java:165]: True but doesn't harm to have comments though, doesn't make it any harder to follow

**Comment by t92549** [comment]
There is now only this that remains with no coverage:
https://github.com/gchq/Gaffer/blob/1ef3621323064ab6a15c84fa554d35fe1dea003d/library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertexProperty.java#L113-L116
If I am honest I don't really understand this. It might be good to either test it or remove it?

**Review by t92549 — APPROVED** [review]

**Pull request merged**